neo/io/nwbio.py

```diff
@@ -517,8 +517,10 @@ def _write_segment(self, nwbfile, segment, electrodes):
         for i, signal in enumerate(
                 chain(segment.analogsignals, segment.irregularlysampledsignals)):
             assert signal.segment is segment
-            signal.name = "%s %s %i" % (signal.name, segment.name, i)
-            if not signal.name:
+            if hasattr(signal, 'name'):
+                signal.name = "%s %s %i" % (signal.name, segment.name, i)
+                logging.warning("Warning signal name exists. New name: %s" % (signal.name))
+            else:
                 signal.name = "%s : analogsignal%d %i" % (segment.name, i, i)
```

Comment:
Is it intentional that you compose the names differently for existing and non-existing name attributes? Wouldn't it make sense to always use a 'segment signal idx' ordering in the name?

Reply:
and the same of course also for events.

Reply:
Yes, that was intentional as I ran into problems with the signal name and the event name when testing with NWB files.